packaging typescript chaincode fails

The chaincode is compiled properly, i.e. shows up in the `dist/` directory but not actually packaged. No error are generated.

## Expected Behavior

Packaged smart contract should show up in the »smart contracts« pane.

## Actual Behavior

When using typescript as the language for the chaincode, nothing happens.

When using javascript as the language for the chaincode, the package shows up.

## Possible Fix

N/A

## Steps to Reproduce

1. Go to Smart contracts pane
2. Select »Create New Project«
3. Create new CRUD project using typescript
4. Project skeleton is created
5. Open new crud project
6. Try to »Package open project«

## Existing issues
<!-- Have you searched for any existing issues or are there any similar issues that you've found? -->
- [x] [Stack Overflow issues](https://stackoverflow.com/questions/tagged/ibp-vscode-extension)
- [x] [GitHub Issues](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues)

<!-- Please include any links to issues here -->

N/A

## Log File

Logfiles are error free. Only says `[INFO] packageSmartContract`

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* IBM Blockchain Platform Extension Version: 1.0.26, 1.0.25, 1.0.24
* Visual Studio Code version: 1.44.1
* Environment name and version (e.g. Node.js v8.12.0, npm v6.4.1): node v10.19.0, npm 6.13.4
* Operating System and version: Darwin x64 19.4.0

I have the same problem [#2020](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/2200#issuecomment-617090178). And when I try to "Package Open Project" I have this error:
> [ERROR] TypeError: Cannot read property 'fsPath' of undefined

I have the same problem, when I try to "Package Open Project" a sample go CRUE contract project, I get fail log:
```
[2020/4/22 下午12:49:28] [INFO] packageSmartContract
[2020/4/22 下午12:49:28] [ERROR] TypeError: Cannot read property 'fsPath' of undefined
```

I could recreate the packaging failing to do anything after compiling the TypeScript, using VS Code 1.44.x. I'm unable to recreate the `[ERROR] TypeError: Cannot read property 'fsPath' of undefined` error though.

As @johndacost mentioned, a workaround is to downgrade VS Code to a previous version such as 1.39 - until we figure out what has changed in VS Code and what changes we can make in the extension.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,8 @@ Fixes
 
 * Fixed "Cannot read property 'major' of null" error on activation [#2200](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/2200).
 
+* Fixed TypeScript contract packaging failing on VS Code 1.44.x [#2193](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/2193).
+
 
 ![conga comic](https://congacomic.github.io/assets/img/blockheight-71.jpg)
 

--- a/packages/blockchain-extension/RELEASE-NOTES.md
+++ b/packages/blockchain-extension/RELEASE-NOTES.md
@@ -17,6 +17,8 @@ Fixes
 
 * Fixed "Cannot read property 'major' of null" error on activation [#2200](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/2200).
 
+* Fixed TypeScript contract packaging failing on VS Code 1.44.x [#2193](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/2193).
+
 
 This release's Conga Comic:
 ---

--- a/packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts
+++ b/packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts
@@ -309,13 +309,11 @@ async function buildWorkspace(workspaceDir: vscode.WorkspaceFolder): Promise<voi
     // If we have a set of build tasks, then execute the first one.
     if (buildTasks.length > 0) {
         const buildTask: vscode.Task = buildTasks[0];
-        const buildTaskExecution: vscode.TaskExecution = await vscode.tasks.executeTask(buildTask);
+        await vscode.tasks.executeTask(buildTask);
         await new Promise((resolve: any): any => {
-            const buildTaskListener: vscode.Disposable = vscode.tasks.onDidEndTask((e: vscode.TaskEndEvent) => {
-                if (e.execution === buildTaskExecution) {
-                    buildTaskListener.dispose();
-                    resolve();
-                }
+            const buildTaskListener: vscode.Disposable = vscode.tasks.onDidEndTask((_e: vscode.TaskEndEvent) => {
+                buildTaskListener.dispose();
+                resolve();
             });
         });
     }

--- a/packages/blockchain-extension/test/commands/packageSmartContractCommand.test.ts
+++ b/packages/blockchain-extension/test/commands/packageSmartContractCommand.test.ts
@@ -234,7 +234,6 @@ describe('packageSmartContract', () => {
                 onDidEndTaskEventEmitter.fire({ execution: testExecution });
                 onDidEndTaskEventEmitter.fire({ execution: buildExecution });
             }, 1);
-            return buildExecution;
         });
 
     });

